backend/windmill-queue/src/jobs.rs

```diff
@@ -531,133 +533,118 @@ pub async fn add_completed_job<T: Serialize + Send + Sync + ValidableJson>(
+                , result
+                , raw_code
+                , raw_lock
+                , canceled
+                , canceled_by
+                , canceled_reason
+                , job_kind
+                , schedule_path
+                , permissioned_as
+                , flow_status
+                , raw_flow
+                , is_flow_step
+                , is_skipped
+                , language
+                , email
+                , visible_to_owner
+                , mem_peak
+                , tag
+                , priority
+                )
+            SELECT 
+                  queue.workspace_id
+                , queue.id
+                , queue.parent_job
+                , queue.created_by
+                , queue.created_at
+                , queue.started_at
+                , COALESCE($2::bigint, (EXTRACT('epoch' FROM (now())) - EXTRACT('epoch' FROM (COALESCE(queue.started_at, now())))) * 1000)
+                , $3 AS success
+                , queue.script_hash
+                , queue.script_path
+                , queue.args
+                , $4 AS result
+                , queue.raw_code
+                , queue.raw_lock
+                , queue.canceled OR COALESCE($7, queue.canceled_by) IS NOT NULL as canceled
+                , COALESCE($7, queue.canceled_by) as canceled_by
+                , COALESCE($8, queue.canceled_reason) as canceled_reason
+                , queue.job_kind
+                , queue.schedule_path
+                , queue.permissioned_as
+                , queue.flow_status
+                , queue.raw_flow
+                , queue.is_flow_step
+                , $5 AS is_skipped
+                , queue.language
+                , queue.email
+                , queue.visible_to_owner
+                , GREATEST($6, queue.mem_peak) AS mem_peak
+                , queue.tag
+                , queue.priority
+            FROM queue
+            WHERE queue.id = $1
+            LIMIT 1
+            ON CONFLICT (id) DO UPDATE SET 
+                success = $3, 
+                result = $4
+            RETURNING duration_ms, canceled
+            "#,
+            /* $1 */ queued_job.id,
+            /* $2 */ duration,
+            /* $3 */ success,
+            /* $4 */ result as Json<&T>,
+            /* $5 */ skipped,
+            /* $6 */ if mem_peak > 0 { Some(mem_peak) } else { None },
+            /* $7 */ canceled_by,
+            /* $8 */ canceled_reason,
         )
         .fetch_one(&mut *tx)
         .await
+        .map(|record| (record.duration_ms, record.canceled))
         .map_err(|e| Error::InternalErr(format!("Could not add completed job {job_id}: {e:#}")))?;
 
-
+        // TODO(uael): to remove ?
         if !queued_job.is_flow_step {
-            if _duration > 500
-                && (queued_job.job_kind == JobKind::Script
-                    || queued_job.job_kind == JobKind::Preview)
-            {
-                if let Err(e) = sqlx::query!(
```

Comment:
@rubenfiszel Removed this as after this change, `completed_job.flow_status` should have the same value as `q.flow_status`